float64 CurrentX
float64 CurrentY
float64 VelX
float64 VelY
